File: back/MyStore.Core/Domain/Product.cs
using MyStore.Core.Domain.Common;

namespace MyStore.Core.Domain
{
    public class Product : BaseModel
    {
        public string Name { get; set; }

        public decimal Price { get; set; }

        public string? Description { get; set; }

        public override bool Equals(object? obj)
        {
            if (obj is not Product product)
                return false;
            return product.Id == Id
                && product.Name == Name 
                && product.Price == Price
                && product.Description == Description;
        }
    }
}

File: front/src/app/_models/product.ts
import { BaseModel } from "./base";

export class Product extends BaseModel {
  name: string;
  price: number;
  description: string;
  shippingDate: Date;
  shipped: boolean;

  constructor(jsonData: any){
    super(jsonData);
    this.name = jsonData.name;
    this.price = jsonData.price;
    this.description = jsonData.description;
    this.shippingDate = jsonData.shippingDate;
    this.shipped = jsonData.shipped === `true`;
  }
}

/*
Copyright Google LLC. All Rights Reserved.
Use of this source code is governed by an MIT-style license that
can be found in the LICENSE file at https://angular.io/license
*/